docs/reference/configuration/Storage.md

```diff
@@ -94,3 +96,97 @@ Name: nvstore.max_keys
     Macro name: NVSTORE_MAX_KEYS
     Value: 16 (set by library:nvstore)
 ```
+
+### BlockDevice - default configuration
+
+The Mbed OS configuration allows you to add block devices as components using the `targets.json` file or target overrides in the application configuration file.
+
+When one of the following components is enabled, a default block device is set in the system:
+
+1. SPIF component.
+1. DATAFLASH component.
+1. SD component.
+
+Components can coexist in the system. A device can have SPIF and SD or any combination of block devices enabled but only one default block device.
+
+The list above is in the order of precedence shows which block device is the default one if more than one component is enabled.
+
+#### Configuring components
+
+Adding "components": ["???"] in targets.json:
+
+```
+    "K64F": {
+        "supported_form_factors": ["ARDUINO"],
+        "components": ["SD"],
+        "core": "Cortex-M4F",
+        "supported_toolchains": ["ARM", "GCC_ARM", "IAR"],
+        "extra_labels": ["Freescale", "MCUXpresso_MCUS", "KSDK2_MCUS", "FRDM", "KPSDK_MCUS", "KPSDK_CODE", "MCU_K64F", "Freescale_EMAC"],
+        "is_disk_virtual": true,
+        "macros": ["CPU_MK64FN1M0VMD12", "FSL_RTOS_MBED"],
+        "inherits": ["Target"],
+        "detect_code": ["0240"],
+        "device_has": ["USTICKER", "LPTICKER", "RTC", "CRC", "ANALOGIN", "ANALOGOUT", "EMAC", "I2C", "I2CSLAVE", "INTERRUPTIN", "PORTIN", "PORTINOUT", "PORTOUT", "PWMOUT", "SERIAL", "SERIAL_FC", "SERIAL_ASYNCH", "SLEEP", "SPI", "SPI_ASYNCH", "SPISLAVE", "STDIO_MESSAGES", "STORAGE", "TRNG", "FLASH"],
+        "features": ["STORAGE"],
+        "release_versions": ["2", "5"],
+        "device_name": "MK64FN1M0xxx12",
+        "bootloader_supported": true,
+        "overrides": {
+            "network-default-interface-type": "ETHERNET"
+        }
+    },
+```
+
+Adding "target.components_add": ["???"] in application config file:
```

Comment:
@theotherjimmy As it's the weekend in Israel, could you please tell me what should go here?

Reply:
Same as above.

Reply:
So could I change it to

Adding "target.components_add": ["<SD>"] in application config file:

For clarity?

Reply:
No. That particular line is somewhat nonsense, somewhat leads to unexpected behavior.

Reply:
Woof. Alright.

"The following `mbed_app.json` snippet enables the Spi flash component when compiling for the MTB_ADV_WISE_1570 target.

```
       "MTB_ADV_WISE_1570": {
            "target.components_add": ["SPIF"]
       }
```

(removed unrelated garbage from the example)

Reply:
@theotherjimmy Let's keep the comments professional and kind. I know @yossi2le and I both appreciate your help streamlining the example, but it doesn't mean the redundant lines are "garbage".

Reply:
As in "things to be thrown out"? I could have picked a more constructive word for sure. That word may have been in poor taste. Edited.

Personally I think unrelated lines take away from an example as they have the potential to distract or misdirect the reader and I find that having them is a determent to the example.

I meat no ill will or disparaging remarks to anyone or their writing skills/style.

Reply:
@theotherjimmy No hard feelings;)